tests/users/test_dao.py

```diff
@@ -148,6 +150,150 @@ def test_dao_delete_only_user_admin(self):
             (messages.USER_CANT_DELETE, HTTPStatus.BAD_REQUEST), dao_result
         )
 
+    def test_dao_delete_user(self):
+        dao = UserDAO()
+
+        user = UserModel(
+            name=user2["name"],
+            email=user2["email"],
+            username=user2["username"],
+            password=user2["password"],
+            terms_and_conditions_checked=user2["terms_and_conditions_checked"],
+        )
+        db.session.add(user)
+        db.session.commit()
+
+        # Verify that user was inserted in database through DAO
+        before_delete_user = UserModel.query.filter_by(id=2).first()
+        self.assertIsNotNone(before_delete_user)
+
+        # Verify email
+        token = generate_confirmation_token(user2["email"])
+        result = dao.confirm_registration(token)
+        self.assertTrue(user.is_email_verified)
+
+        dao_result = dao.delete_user(2)
+
+        after_delete_user = UserModel.query.filter_by(id=2).first()
+        self.assertIsNone(after_delete_user)
+        self.assertEqual(
+            (messages.USER_SUCCESSFULLY_DELETED, HTTPStatus.OK), dao_result
+        )
+
+    def test_dao_delete_user_that_does_not_exist(self):
+        dao = UserDAO()
+
+        # Verify that user does not exist
+        before_delete_user = UserModel.query.filter_by(id=2).first()
+        self.assertIsNone(before_delete_user)
+
+        dao_result = dao.delete_user(2)
+
+        after_delete_user = UserModel.query.filter_by(id=2).first()
+        self.assertEqual(
+            (messages.USER_DOES_NOT_EXIST, HTTPStatus.NOT_FOUND), dao_result
```

Comment:
![ezgif com-gif-maker (70)](https://user-images.githubusercontent.com/29667122/123968253-69804f00-d9fa-11eb-952b-50a3bf885bd3.gif)

Reply:
Wow I understand now. @mtreacy002 thank you so much for the work you put in adding this here 👏🏾 🎉 
I wasn't understanding, in the beginning, why, and now I understand why that it of code was removed.